e2e/gui/src/test/java/com/epam/pipeline/autotests/LaunchClusterTest.java

```diff
@@ -205,4 +230,309 @@ public void autoScaledClusterWithDefaultChildNodesValidationTest() {
+                .ensure(log(), containsMessages("The following jobs cannot be satisfied with the " +
+                        "requested resources and therefore they will be rejected: 1 (150 cpu)"))
+                .ssh(shell -> shell
+                        .execute("qsub -b y -pe local 50 sleep 5m")
+                        .close());
+        navigationMenu()
+                .runs()
+                .activeRuns()
+                .showLog(getRunId())
+                .waitForNestedRunsLink()
+                .clickOnNestedRunLink()
+                .ensure(STATUS, text(String.valueOf(Integer.parseInt(getRunId()) + 1)));
+    }
+
+    @Test
+    @TestCase({"EPMCMBIBPC-3159"})
+    public void hybridAutoScaledClusterCPUDeadlockWithAdditionalRestrictions() {
+        final String systemParam = "CP_CAP_AUTOSCALE_HYBRID_MAX_CORE_PER_NODE";
+        library()
+                .createPipeline(Template.SHELL, getPipelineName())
+                .clickOnPipeline(getPipelineName())
+                .firstVersion()
+                .runPipeline()
+                .setDefaultLaunchOptions()
+                .enableClusterLaunch()
+                .clusterSettingsForm(autoScaledSettingForm)
+                .enableHybridClusterSelect()
+                .click(button("OK"))
+                .click(START_IDLE)
+                .clickAddSystemParameter()
+                .selectSystemParameters(systemParam)
+                .inputSystemParameterValue(systemParam, "40")
+                .launch(this)
+                .shouldContainRun(getPipelineName(), getRunId())
+                .showLog(getRunId())
+                .waitForSshLink()
+                .ssh(shell -> shell
+                        .execute("qsub -b y -pe local 50 sleep 5m")
+                        .assertOutputContains("Your job 1 (\"sleep\") has been submitted")
+                        .sleep(20, SECONDS)
+                        .execute("qstat")
+                        .close());
+        navigationMenu()
+                .runs()
+                .activeRuns()
+                .showLog(getRunId())
+                .waitForTask(gridEngineAutoscalingTask)
+                .click(taskWithName(gridEngineAutoscalingTask))
+                .ensure(log(), containsMessages("The following jobs cannot be satisfied with the " +
+                        "requested resources and therefore they will be rejected: 1 (50 cpu)"));
+    }
+
+    @Test
+    @TestCase({"EPMCMBIBPC-3156"})
+    public void autoScaledClusterWorkersPriceTypeSpot() {
+        library()
+                .createPipeline(Template.SHELL, getPipelineName())
+                .clickOnPipeline(getPipelineName())
+                .firstVersion()
+                .runPipeline()
+                .setDefaultLaunchOptions()
+                .enableClusterLaunch()
+                .clusterSettingsForm(autoScaledSettingForm)
+                .setWorkingNodesCount("1")
+                .setWorkersPriceType("Spot")
```

Comment:
We should set the 'e2e.ui.spot.price.name' value for all providers

Reply:
The same for all hard coded "Spot" value

Reply:
Done